objects/tag.c

```diff
@@ -220,16 +219,32 @@ is_client_tagged(client_t *c, tag_t *t)
     return false;
 }
 
-/** Get the index of the first selected tag.
- * \return Its index.
+/** Get the index of the tag with focused client or first selected 
+ * \return Its index
  */
 int
-tags_get_first_selected_index(void)
+tags_get_current_or_first_selected_index(void)
 {
-    foreach(tag, globalconf.tags)
+    /* Consider "current desktop" a tag, that has focused window,
+     * basically a tag user actively interacts with.
+     * If no focused windows are present, fallback to first selected.
+     */
+     tag_t **first_tag = NULL;
+     client_t *client = globalconf.focus.client;
+     foreach(tag, globalconf.tags)
+     {
         if((*tag)->selected)
-            return tag_array_indexof(&globalconf.tags, tag);
-    return 0;
+        {
+            if(!first_tag)
+                first_tag = tag;
+            for(int i = 0; i < (*tag)->clients.len; i++)
+            {
+                if((*tag)->clients.tab[i] == client)
+                    return tag_array_indexof(&globalconf.tags, tag);
+            }
+        }
+     }
+     return tag_array_indexof(&globalconf.tags, first_tag);
```

Comment:
I would suggest:
```C
if (globalconf.focus.client)
{
    foreach(tag, globalconf.tags)
    {
        if((*tag)->selected && is_client_tagged(globalconf.focus.client, *tag))
            return tag_array_indexof(&globalconf.tags, tag);
    }
}
foreach(tag, globalconf.tags)
{
    if ((*tag)->selected)
        return tag_array_indexof(&globalconf.tags, tag);
}
return 0;
```
I consider this easier to read (I can immediately see that it first tries to find a tag for the focused client, then just any selected tag) and it also works "correctly" if no tag at all is currently selected.